samples/python/pbapi/__init__.py

```diff
@@ -232,7 +232,10 @@ def list_beacons(self, arguments):
                     .format(next_page_token, beacons_resp['nextPageToken'])
             if next_page_token is not None and beacons_resp['nextPageToken'] == next_page_token:
                 break
-            next_page_token = beacons_resp['nextPageToken']
+            try: 
+                next_page_token = beacons_resp['nextPageToken']
+            except KeyError:
+                break
```

Comment:
I suppose you should consume the beacons even if there is no next page token, no?

Reply:
Ah, looks like this logic was even more messed up than that. Fixed, I think.